pkg/collector/collector.go

```diff
@@ -5,10 +5,21 @@ type Collector interface {
 	Name() string
 }
 
+type VersionCollector interface {
+	GetServerVersion() (string, error)
+}
+
 type commonCollector struct {
 	name string
 }
 
+func newCommonCollector(name string) *commonCollector {
+	col := &commonCollector{
```

Comment:
do we need col? Seems like:

```
return &commonCollector{
                    name: name
}
```
 would be fine

Reply:
Done.